src/WasmModule.ts

```diff
@@ -0,0 +1,236 @@
+import { decode } from '@webassemblyjs/wasm-parser';
+import ast from '@webassemblyjs/ast';
+
+import Graph from './Graph';
+import { IdMap, ModuleJSON } from './rollup/types';
+
+import { makeLegal } from './utils/identifierHelpers';
+import { basename, extname } from './utils/path';
+
+import ExternalModule from './ExternalModule';
+import Module from './Module';
+
+import ModuleScope from './ast/scopes/ModuleScope';
+import Variable from './ast/variables/Variable';
+import NamespaceVariable from './ast/variables/NamespaceVariable';
+
+const decoderOpts = {
+	ignoreCodeSection: true,
+	ignoreDataSection: true
+};
+
+type BuildLoadArgs = {
+	NAME: string;
+	URL: string;
+	IMPORT_OBJECT?: string;
+};
+
+// FIXME(sven): uncommented the loader you need for your current env
+// how to get the env?
```

Comment:
That was an attempt for adding tests. How can I get access to the env i'm in? I guess I could use the finalizers for that?

Also sometime I see `__dirname is not defined` when running the tests.